frontend/pages/projects/[projectId].js

```diff
@@ -128,6 +129,24 @@ export default function ProjectPage({
 
   const smallScreenSize = useMediaQuery((theme) => theme.breakpoints.down("sm"));
 
+  // Handle remove bell icon notification
+  const { notifications, setNotificationsRead, refreshNotifications } = useContext(UserContext);
+  const handleReadNotifications = async (notificationType) => {
+    const notification_to_set_read = notifications.filter(
+      (n) =>
+        n.notification_type === notificationType &&
+        n.project.url_slug === project.url_slug
+    );
+    await setNotificationsRead(token, notification_to_set_read, locale);
+    await refreshNotifications();
+  };
+
+  useEffect(async () => {
+    await handleReadNotifications(NOTIFICATION_TYPES.indexOf("org_project_published"));
+  }, [notifications.length !== 0]); /* end of removing bell icon notification
```

Comment:
I think using useEffect is actually fine here. In `handleReadNotifications` I'd make sure that `setNotificationsRead` and `refreshNotifications` are only run when there are actually notification_to_set_read.

```suggestion
  }, []); /* end of removing bell icon notification
```

Reply:
I had this problem initially, that using this suggestion would call the function handleReadNotifications, but this wouldn't remove the notification icon. Hence the change to notifications.length !=0. I'm not sure why it doesn't work with [], as you suggested.